src/hotspot/share/nmt/memMapPrinter.cpp

```diff
@@ -172,6 +172,8 @@ static bool vma_touches_thread_stack(const void* from, const void* to, const Thr
   // Very rarely however is a VMA backing a thread stack folded together with another adjacent VMA by the
   // kernel. That can happen, e.g., for non-java threads that don't have guard pages.
   // Therefore we go for the simplest way here and check for intersection between VMA and thread stack.
+  // If the stack size is zero (i.e. no stack assigned yet), then return false.
+  if (t->stack_size() == 0) return false;
```

Comment:
I think this should go in the iteration loop:
```
  for (JavaThreadIteratorWithHandle jtiwh; JavaThread* t = jtiwh.next(); ) {
    // We can encounter new threads whose stack has not yet been initialized - so skip them
    if (t->stack_size() == 0) continue;
    HANDLE_THREAD(t);
  }
```
or even
```
  for (JavaThreadIteratorWithHandle jtiwh; JavaThread* t = jtiwh.next(); ) {
    // We can encounter new threads whose stack has not yet been initialized - so skip them in debug to avoid assertion
    DEBUG_ONLY(if (t->stack_size() == 0) continue;)
    HANDLE_THREAD(t);
  }
```
this assumes `range_intersects` will correctly handle a zero base and end.

Reply:
I would prefer it here, or in HANDLE_THREAD macro, to make sure we do this for all threads printed.

Reply:
I will put it in HANDLE_THREAD to cover the other tests.

Reply:
It is not needed for the other tests - those threads can't be new.

This is an issue with using the JavaThread iterator so should be visibly seen in the itertation loop IMO.